### ISSUE TITLE
E0401: Unable to import 'pyspark.context' (import-error)

### ISSUE THREAD

**Issue by AndreasLuckert** [PyCQA/pylint#5100]
### Bug description

```python
In a python-script I have:
import pyspark.context

-> this causes pylint to fail with 'E0401: Unable to import 'pyspark.context' (import-error)'
```

### Command used

```shell
pylint path/to/file/with/pyspark/import.py
```

### Pylint output

```shell
pylint.........................................................................Failed
- hook id: pylint
- exit code: 2

************* Module job_a2_table_preprocessing_pyspark
src\glue_jobs\job_a2_table_preprocessing_pyspark.py:10:0: E0401: Unable to import 'pyspark.context' (import-error)
```

### Expected behavior

$ pylint src/glue_jobs/job_a2_table_preprocessing_pyspark.py 

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 8.21/10, +1.79)

### Pylint version

```shell
The latest pylint version will be used, which is currently:
$ pylint --version
pylint 2.11.1
astroid 2.8.0
Python 3.9.4 (tags/v3.9.4:1f2e308, Apr  6 2021, 13:40:21) [MSC v.1928 64 bit (AMD64)]

Pylint is then invoked via a precommit-hook like so:

- repo: https://github.com/PyCQA/pylint
    rev: 591a23adcfdd2fe20b8cfdb9e4e07772c8f454f8 # frozen: v2.10.2
    hooks:
      - id: pylint
        # args: ["--exit-zero"]
        args:
          [
            "--fail-under=9.98",
            "--evaluation='10.0 - ((float(5 * error + warning + 0.1 * refactor + convention) / statement) * 10)'",
          ]
```

### OS / Environment

Windows 10

### Additional dependencies

From precommit-hook:

        additional_dependencies:
          # Spellchecking
          - pyenchant
          # Sphinx documentation
          - sphinx_rtd_theme
          - sphinxcontrib-packages
          # ics dependencies
          - awswrangler==2.8.0
          - babel==2.9.1
          - category-encoders==2.2.2
          - joblib==1.0.1
          - markdown==3.3.4
          - matplotlib==3.3.4
          - openpyxl==3.0.7
          - pandas==1.1.5
          - pyarrow==4.0.0
          - pytz==2021.1
          - requests==2.25.1
          - scikit-learn==0.24.2
          - simple_salesforce==1.11.1

**Comment by DanielNoord** [comment]
Is this `pre-commit` call done in your CI or in your local environment?
If it is the former, is `pyspark` actually part of any of the dependencies installed in that CI? It is not included in the `precommit-hook` you provided and might need to be installed.
If it is the latter, I don't know the exact behaviour of `pre-commit` but you might still need to add `pyspark` to the dependencies of `pre-commit`. I think it should just use your local environment, but I'm not 100% sure. If it is indeed your local environment, could you post the output of `pip freeze`?

**Comment by mbyrnepr2** [comment]
I think it's reasonable to close this. As Daniël suggests, the error is emitted if pyspark is not installed in the same environment where pylint is run. We can always re-open if we obtain further information.
